R/extract.R

```diff
@@ -88,3 +88,13 @@ str_extract <- function(x, into, regex, convert = FALSE) {
 
   out
 }
+
+str_match_first <- function(x, regex) {
+  if (length(x) == 0) {
+    # Can't determine number of matches
```

Comment:
I think this needs a little extra thought (I added the branch just as a placeholder:

``` r

stringi::stri_match_first(character(), regex = "(.)-(.)")
#>      [,1] [,2] [,3]
tidyr:::str_match_first(character(), regex = "(.)-(.)")
#> NULL
```

Reply:
Hmmmm, also problematic when there are no matches:

``` r
str(stringi::stri_match_first("", regex = "(.)-(.)"))
#>  chr [1, 1:3] NA NA NA
str(tidyr:::str_match_first("", regex = "(.)-(.)"))
#>  chr[1, 0 ]
```

Reply:
Just remembered I have some code lying around for exactly this problem 😀